content/docs/languages-sdks/python/_index.md

```diff
@@ -29,8 +29,38 @@ with examples available in Python. These concepts are made available to you in t
 The Pulumi SDK is available to Python developers as a package distributed on PyPI. To learn more,
 [refer to the Pulumi SDK Reference Guide](/docs/reference/pkg/python/pulumi/).
 
+### Inputs and Outputs
+
 The Pulumi programming model includes a core concept of `Input` and `Output` values, which are used to track how outputs of one resource flow in as inputs to another resource.  This concept is important to understand when getting started with Python and Pulumi, and the [Inputs and Outputs](/docs/concepts/inputs-outputs/) documentation is recommended to get a feel for how to work with this core part of Pulumi in common cases.
 
+In Python, inputs that are objects, that is inputs that group multiple values together, can be reprented either as classes or as dictionary literals. The types for the argument classes have the suffix `Args`, whereas the types for the dictionaries have the suffix `ArgsDict`. Both types take the same arguments, but the dictionary types are often more concise.
+
+{{% notes type="info" %}}
+The types with the suffix `ArgsDict` for dictionary literals were introduced in July 2024. You can still use dictionary literals with providers that have not been updated yet with this change, but you will not benefit from the type checking that the new types provide.
```

Comment:
It might be worth linking to a doc that describes what a provider is, for anyone new to Pulumi still trying to understand how it works.

```suggestion
The types with the suffix `ArgsDict` for dictionary literals were introduced in July 2024. You can still use dictionary literals with [providers](/docs/concepts/how-pulumi-works/#resource-providers) that have not been updated yet with this change, but you will not benefit from the type checking that the new types provide.
```